can_testbench.py

```diff
@@ -500,144 +777,279 @@ class RxMessageLayout(MessageLayout):
+                msgTable.signalGraphedChanged.connect(self.onSignalGraphedChanged)
                 self.msgTableDict[msg.message.frame_id] = msgTable
             tabLayout.addWidget(msgLayout)
+            
+            if label:
+                msgLayout.setInfoLabel(label)
+                label = None
+        
+        layout = QVBoxLayout(tab)  # This is the layout for the tab itself
+        layout.addWidget(scrollArea)  # Add the scrollArea to the tab's layout
+
+        self.tabs.add(tab)
+        tabWidget.addTab(tab, title)
+        tabWidget.setTabWhatsThis(tabWidget.count() - 1 ,self.channel)
+        
+    def shutdown(self):
+        self.canBus.shutdown()
+        for tab in self.tabs:
+            tab.deleteLater()
+        for graph in self.graphWindows:
+            graph.deleteLater()
+        
 
+class MainApp(QMainWindow):
+    """
+    A class that represents the main application
+
+    Attributes:
+    config (CanConfig): Source of truth for config settings
+    configLayout (ConfigLayout): Initial tab for user to set config
+    dbcDb (Database): Message data for loaded DBC file
+    openCans (Dict): All connected CANs and their associated channel
+    """
+    def __init__(self):
+        super().__init__()
+        self.setWindowTitle('CAN Testbench')
+        
+        self.config = CanConfig()
+        self.configLayout = ConfigLayout(self.config)
+        self.configLayout.dbcOpened.connect(self.openDbc)
+        self.configLayout.connectPressed.connect(self.connectCan)
+        
+        self.dbcDb = None
+        self.openCans = {}
+        self.initUI()
+        self.resizeToScreenFraction()
+
+    def resizeToScreenFraction(self, fractionWidth=1.0, fractionHeight=0.8):
+        # Get the screen size
+        screen = QApplication.primaryScreen()
+        screenSize = screen.size()
+
+        # Calculate the window size as a fraction of the screen size
+        newWidth = int(screenSize.width() * fractionWidth)
+        newHeight = int(screenSize.height() * fractionHeight)
+        newWidth = min(newWidth,1350)
+        logging.debug(f'Window size: {newWidth}x{newHeight}')
+
+        # Resize the window
+        self.resize(newWidth, newHeight)
+    
+    def setupLaunchTab(self):
+        tab = QWidget()
+
+        scrollContent = QWidget()
+        scrollArea = QScrollArea(tab)
+        scrollArea.setWidgetResizable(True)
         scrollArea.setWidget(scrollContent)
+    
+        tabLayout = QVBoxLayout(scrollContent)
+        tabLayout.addWidget(self.configLayout)
+        
         layout = QVBoxLayout(tab)  # This is the layout for the tab itself
         layout.addWidget(scrollArea)  # Add the scrollArea to the tab's layout
 
-        self.tabWidget.addTab(tab, title)
-
+        self.tabWidget.addTab(tab, 'CAN Config')
+        tabBar = self.tabWidget.tabBar()
+        tabBar.tabButton(0, QTabBar.ButtonPosition.RightSide).resize(0, 0)
```

Comment:
This is failing for me when I try to run it.
  File "/Users/jerrymailloux/dev/can_testbench/can_testbench.py", line 976, in setupLaunchTab
    tabBar.tabButton(0, QTabBar.ButtonPosition.RightSide).resize(0, 0)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'resize'

Reply:
Should be fixed now, I now check both sides and check that the button exists before calling resize.